### PR TITLE
Tidy-Up CSCC API

### PR DESCRIPTION
**Pull request by blueandgold**
This is the follow-up to tidy up the initial PR #1746 that added the working changes for CSCC API.

I will add the unit tests after the initial review by joe and andrew.  I would appreciate any feedback on making the Create Mix-In to be as generic as possible.

### CODE REVIEW

File: google/cloud/forseti/common/gcp_api/securitycenter.py [hunk 1]
```diff
@@ -74,37 +69,51 @@ class _SecurityCenterOrganizationsFindingsRepository(
...
+            )
+            LOGGER.debug('Created finding response in CSCC: %s', list(response))
         except (errors.HttpError, HttpLib2Error) as e:
-            LOGGER.info('>>>>> %s', e)
-            #if _is_status_not_found(e):
-            #    return []
-            #raise api_errors.ApiExecutionError(project_id, e)
-            raise api_errors.ApiExecutionError(e)
+            LOGGER.error(
+                'Unable to create resource:\n%s\n'
```
Comment (joecheuk) [inline google/cloud/forseti/common/gcp_api/securitycenter.py]: Maybe 'Unable to create CSCC finding:' is easier to understand in this case?

File: google/cloud/forseti/common/gcp_api/repository_mixins.py [hunk 2]
```diff
@@ -291,24 +286,20 @@ def search(self, query=None, fields=None, max_results=500, verb='search'):
...
-                     'orgName': 'organizations/' + str(organization_id)}
-        if query:
-            req_body['sourceFinding'] = query
-
-#        req_body[self._max_results_field] = max_results
+        arguments = {}
+        if kwargs.get('arguments'):
+            arguments.update(kwargs.get('arguments'))
 
         for resp in self.execute_query(
```
[the remark this answers is not in the document]
Comment (blueandgold) [reply google/cloud/forseti/common/gcp_api/repository_mixins.py]: Good catch, done.

File: google/cloud/forseti/common/gcp_api/securitycenter.py (hunk 1 above)
Comment (blueandgold) [reply google/cloud/forseti/common/gcp_api/securitycenter.py]: Done

File: configs/server/forseti_conf_server.yaml.in [hunk 3]
```diff
@@ -352,6 +352,8 @@ notifier:
     violation:
       cscc:
         enabled: false
+        mode: api
```
[the remark this answers is not in the document]
Comment (blueandgold) [reply configs/server/forseti_conf_server.yaml.in]: I think there is a need to support both modes, to give users time to transition to the new api.

I think also, that having an explicit field to denote the `mode` is more clear and understandable, which then is easier to maintain.  So, if no strong objection, would prefer to leave it here.

File: google/cloud/forseti/notifier/notifiers/cscc_notifier.py [hunk 4]
```diff
@@ -142,33 +137,43 @@ def _transform_for_cscc_api(self, violations):
...
-                organization_id=organization_id,
-                finding=finding
-            )
+            LOGGER.debug('Creating finding CSCC:\n%s.', finding)
+            try:
+                client.create_finding(organization_id, finding)
+                LOGGER.debug('Successfully created finding in CSCC:\n%s',
+                             finding)
+            except api_errors.ApiExecutionError:
+                continue
```
[the remark this answers is not in the document]
Comment (blueandgold) [reply google/cloud/forseti/notifier/notifiers/cscc_notifier.py]: Done.

**Pull request merged**